tests/integrationv2/test_hello_retry_requests.py

```diff
@@ -0,0 +1,160 @@
+import copy
+import os
+import pytest
+import time
+
+from configuration import available_ports, TLS13_CIPHERS, ALL_TEST_CURVES, ALL_TEST_CERTS, PROVIDERS, PROTOCOLS
+from common import Certificates, ProviderOptions, Protocols, data_bytes, Ciphers, Curves
+from fixtures import managed_process
+from providers import Provider, S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name
+
+def verify_hello_retry_request(server):  
+    marker_found = False
+    client_hello_count = 0
+    server_hello_count = 0 
+    finished_count = 0
+    marker = b"cf 21 ad 74 e5 9a 61 11 be 1d"
+
+    for results in server.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+
+        if marker in results.stdout:
```

Comment:
You can also verify that `bytes_to_send` exists in the server output

Reply:
Good catch, will add.